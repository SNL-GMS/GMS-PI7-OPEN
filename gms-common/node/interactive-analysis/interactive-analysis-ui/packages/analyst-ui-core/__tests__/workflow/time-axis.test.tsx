import { ReactWrapper } from 'enzyme';
import * as React from 'react';
import {
  WorkflowTimeAxis,
  WorkflowTimeAxisProps
} from '~analyst-ui/components/workflow/components/workflow-time-axis';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');
// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Adapter = require('enzyme-adapter-react-16');

const mockWorkflowTimeAxisProps: WorkflowTimeAxisProps = {
  startTimeSecs: 3000,
  endTimeSecs: 3600,
  intervalDurationSecs: 60,
  maxWidth: '400',
  pixelsPerHour: 5
};

describe('workflow time axis tests', () => {
  beforeEach(() => {
    Enzyme.configure({ adapter: new Adapter() });
  });

  // Not the most useful test because the d3 does not render
  it('renders a snapshot', (done: jest.DoneCallback) => {
    const wrapper: ReactWrapper = Enzyme.mount(
      <WorkflowTimeAxis {...mockWorkflowTimeAxisProps} />
    );

    setImmediate(() => {
      wrapper.update();
      expect(wrapper.find(WorkflowTimeAxis))
        .toMatchSnapshot('component');

      done();
    });
  });
});
