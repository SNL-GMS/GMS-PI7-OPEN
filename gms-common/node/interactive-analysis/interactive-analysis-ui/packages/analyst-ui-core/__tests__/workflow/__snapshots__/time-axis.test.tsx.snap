// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workflow time axis tests renders a snapshot: component 1`] = `
<WorkflowTimeAxis
  endTimeSecs={3600}
  intervalDurationSecs={60}
  maxWidth="400"
  pixelsPerHour={5}
  startTimeSecs={3000}
>
  <div
    className="time-axis-wrapper"
    style={
      Object {
        "maxWidth": "400",
      }
    }
  >
    <div
      className="time-axis"
    />
  </div>
</WorkflowTimeAxis>
`;
